fcrepo4/fcrepo4
# PR is a two in one:  fcrepo-2602 and fcrepo-2603

ActivityStreams JSON is now used for message bodies of transmitted events.  
The new ActivityStreams based JSON-LD uses  proper actor values for emitted messages

# Resolves:
https://jira.duraspace.org/browse/FCREPO-2602
   https://jira.duraspace.org/browse/FCREPO-2603

# What does the PR do? 

This commit retools the message bodies to use the ActivityStreams vocabulary.  In addition to reworking the body by enabling the -Dfcrepo.auth.webac.userAgent.baseUri=http://your-user-agent-base-uri/ flag on your fcrepo4 instance you can ensure that the value of the "actor" property will conform to the URLs used by SOLID WebAC. 

# What's new?
Since the format of the message body changed, any consumers of the messages will need to be retooled accordingly.

# How should this be tested?
Unit tests cover the serialization and deserialization processes. If you would like to verify the contents of the generated messages you can use the following wiki to view the events generated by fedora:
https://wiki.duraspace.org/display/FEDORA4x/How+to+Inspect+Event+Messages+Generated+by+Fedora
If you do not use the -Dfcrepo.auth.webac.userAgent.baseUri flag the actor value should appear as "#<username>".
# Additional Notes:
* Breaking change:  Downstream consumers of messages will need to be updated to work with new message format.
* [Documentation will need to be updated] (https://jira.duraspace.org/browse/FCREPO-2606)
# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers
